### PR TITLE
copr: support extract scalar value from bit

### PR DESCRIPTION
**Pull request by Yisaer**
### What is changed and how it works?

Issue Number: close https://github.com/tikv/tikv/issues/11893

What's Changed:

In https://github.com/tikv/tikv/pull/11968 ,  we supported push bit cloumn down but we forgot to support mysqlBit during extract scalar value.

This pr fixes it.

```commit-message
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by Yisaer** [comment]
/cc @skyzh @zhongzc 

And I don't know how to design unit testcase for this,  could you help me? @skyzh

**Review by zhongzc — APPROVED** [review]

File: components/tidb_query_expr/src/types/expr_builder.rs [hunk 1]
```diff
@@ -395,6 +398,16 @@ fn extract_scalar_value_int64(val: Vec<u8>) -> Result<ScalarValue> {
     Ok(ScalarValue::Int(Some(value)))
 }
 
+#[inline]
+fn extract_scalar_value_uint64_from_bits(val: Vec<u8>) -> Result<ScalarValue> {
+    debug_assert!(val.len() <= 8);
+    let mut res = 0;
+    for (i, v) in val.iter().enumerate() {
+        res |= (*v as u64) << ((val.len() - i - 1) * 8);
+    }
```
Comment (zhongzc) [inline components/tidb_query_expr/src/types/expr_builder.rs:407]: If I don't think wrong, could the logic to be:
```rust
let mut res = 0;
for v in val {
    res <<= 8;
    res |= v as u64;
}
```

Comment (Yisaer) [reply components/tidb_query_expr/src/types/expr_builder.rs]: right, yours is more clear, updated.

**Review by longfangsong — APPROVED** [review]

**Review by skyzh — APPROVED** [review]

**Pull request merged**